frontend/src/components/Button/index.ts

```diff
@@ -0,0 +1,27 @@
+import { createElement } from "@/utils/dom";
+import "./style.scss";
+
+const template = `
+  <button type="button" class="primary-button"></button>
+`;
+
+export interface IButtonProps {
+  selector?: string;
+  text: string;
+}
+
+export default function Button(props: IButtonProps): HTMLElement {
+  const { selector, text } = props;
+  const $button = createElement(template);
+  $button.innerText = text;
+
+  if (selector) {
+    if (selector[0] === ".") {
+      $button.classList.add(selector);
+    } else if (selector[0] === "#") {
+      $button.id = selector;
+    }
```

Comment:
```suggestion
    if (selector.startsWith(".")) {
      $button.classList.add(selector);
    } else if (selector.startsWith("#")) {
      $button.id = selector;
    }
```

이게 좀 더 직관적이네요.
다만 이렇게 되면 class에 .이 들어가거나 id가 #으로 시작하진 않나요?

Reply:
오 좋은 것 같습니다. 그리고 id나 class에 .이나 # 들어가는걸 생각을 못했네요 !! 수정하도록 하겠습니다